src/Google.Api.Gax/ApiCallRetryExtensions.cs

```diff
@@ -7,89 +7,87 @@
 
 using Google.Protobuf;
 using Grpc.Core;
+using System;
 using System.Threading;
+using System.Threading.Tasks;
 
 namespace Google.Api.Gax
 {
     public static class ApiCallRetryExtensions
     {
-        public static ApiCall<TRequest, TResponse> WithRetry<TRequest, TResponse>(
-            this ApiCall<TRequest, TResponse> originalCall,
-            RetrySettings retrySettings,
-            IClock clock,
-            IScheduler scheduler)
-            where TRequest : class, IMessage<TRequest>
-            where TResponse : class, IMessage<TResponse>
-        {
-            GaxPreconditions.CheckNotNull(originalCall, nameof(originalCall));
-            GaxPreconditions.CheckNotNull(clock, nameof(clock));
+        // By design, the code is mostly duplicated between the async and sync calls.
 
-            if (retrySettings == null)
+        // Async retry
+        public static Func<TRequest, CallSettings, Task<TResponse>> WithRetry<TRequest, TResponse>(
```

Comment:
It seems very odd to me that WithRetry doesn't take any retry settings. Do we definitely need to have this as a separate concept? Given that it's baked into CallSettings, maybe it should be baked into ApiCall as well? I realize there's a *sort* of orthogonality here... but it feels a bit fake somehow.